include/openmc/weight_windows.h

```diff
@@ -0,0 +1,99 @@
+#ifndef OPENMC_WEIGHT_WINDOWS_H
+#define OPENMC_WEIGHT_WINDOWS_H
+
+#include <cstdint>
+#include <unordered_map>
+
+#include <hdf5.h>
+#include <pugixml.hpp>
+
+#include "openmc/constants.h"
+#include "openmc/memory.h"
+#include "openmc/mesh.h"
+#include "openmc/particle.h"
+#include "openmc/vector.h"
+
+namespace openmc {
+
+//==============================================================================
+// Non-member functions
+//==============================================================================
+
+//! Apply weight windows to a particle
+//! \param[in] p  Particle to apply weight windows to
+void apply_weight_windows(Particle& p);
+
+//! Free memory associated with weight windows
+void free_memory_weight_windows();
+
+//==============================================================================
+// Global variables
+//==============================================================================
+
+class WeightWindows;
+
+namespace variance_reduction {
+
+extern std::unordered_map<int32_t, int32_t> ww_map;
+extern vector<unique_ptr<WeightWindows>> weight_windows;
+
+} // namespace variance_reduction
+
+//==============================================================================
+//! Individual weight window information
+//==============================================================================
+
+struct WeightWindow {
+  double lower_weight {-1}; // -1 indicates not valid state
+  double upper_weight {1};
+  double survival_weight {0.5};
+  double weight_cutoff {WEIGHT_CUTOFF};
```

Comment:
It appears we don't actually need to store the weight cutoff in every `WeightWindow`, right? It seems to be a global parameter.